autogoal/kb/_algorithm.py

```diff
@@ -491,7 +511,7 @@ def run(self, a: type_1, b: type_2, ...) -> type_n:
         guaranteed_types = node.output_types
 
         # Here are all the algorithms that could be added new at this point in the graph
-        for algorithm in registry:
+        for algorithm in pool:
```

Comment:
This looks like a very silly mistake on our side, honestly :sweat: so I'm wondering why are there no tests that cover this!